src/value.rs

```diff
@@ -171,186 +175,54 @@ pub enum Date {
-                offset,
-            } => format!(
-                "{:02}:{:02}:{:02}{}",
-                hour,
-                minute,
-                second,
-                offset.format("%z")
-            )
-            .fmt(f),
-            Time::FractionalSecond {
-                hour,
-                minute,
-                second,
-                fractional,
-                offset,
-            } => format!(
-                "{:02}:{:02}:{:02}.{}{:?}",
-                hour,
-                minute,
-                second,
-                fractional.to_string(),
-                offset.format("%z")
-            )
-            .fmt(f),
-        }
-    }
-}
-
-#[derive(Clone, Debug, PartialEq)]
-pub struct TextTimestamp {
-    date: Date,
-    time: Option<Time>,
-}
-
-impl TextTimestamp {
-    pub fn new(date: Date, time: Option<Time>) -> Self {
-        Self { date, time }
-    }
-}
-
 // timestamp - Date/time/timezone moments of arbitrary precision
 // Mostly ISO 8601
 // Enum variant names represent the precision of the variant
-// TODO: Investigate performance impact of large_enum_variant
 #[allow(clippy::large_enum_variant)]
 #[derive(Clone, Debug, PartialEq)]
 pub enum Timestamp {
-    Text(TextTimestamp),
     Year {
-        offset: BigInt,
-        year: BigUint,
+        offset: i32,
+        year: u16,
     },
     Month {
-        offset: BigInt,
-        year: BigUint,
-        month: BigUint,
+        offset: i32,
+        year: u16,
+        month: u8,
     },
     Day {
-        offset: BigInt,
-        year: BigUint,
-        month: BigUint,
-        day: BigUint,
+        offset: i32,
+        year: u16,
+        month: u8,
+        day: u8,
     },
     Minute {
-        offset: BigInt,
-        year: BigUint,
-        month: BigUint,
-        day: BigUint,
-        hour: BigUint,
-        minute: BigUint,
+        offset: i32,
+        year: u16,
+        month: u8,
+        day: u8,
+        hour: u8,
+        minute: u8,
     },
     Second {
-        offset: BigInt,
-        year: BigUint,
-        month: BigUint,
-        day: BigUint,
-        hour: BigUint,
-        minute: BigUint,
-        second: BigUint,
+        offset: i32,
+        year: u16,
+        month: u8,
+        day: u8,
+        hour: u8,
+        minute: u8,
+        second: u8,
     },
     FractionalSecond {
-        offset: BigInt,
-        year: BigUint,
-        month: BigUint,
-        day: BigUint,
-        hour: BigUint,
-        minute: BigUint,
-        second: BigUint,
+        offset: i32,
+        year: u16,
+        month: u8,
+        day: u8,
+        hour: u8,
+        minute: u8,
+        second: u8,
         fraction_coefficient: BigUint,
-        // The restriction of fractional_exponent to i32 rather than BigInt should not pose an issue for any non-pathological use
+        // The restriction of fractional_exponent to u32 rather than BigInt should not pose an issue for any non-pathological use
```

Comment:
This says u32, but `fractional_exponent` is i32

Reply:
Oops, fixed.